library('DESeq2')

#How many genes are differentially expressed by fraction?
dds <- DESeqDataSetFromMatrix(countData = geneCounts, colData = pd, design = ~ Library + Zone + Fetal)
dds <- DESeq(dds)
res <- results(dds)

Irdds <- DESeqDataSetFromMatrix(countData = ribozero.counts, colData = Ribozero, design = ~ Fetal + Zone + Fetal:Zone)
Irdds <- DESeq(Irdds)
Irres <- results(Irdds)
Ipdds <- DESeqDataSetFromMatrix(countData = polya.counts, colData = PolyA, design = ~ Fetal + Zone + Fetal:Zone)
Ipdds <- DESeq(Ipdds)
Ipres <- results(Ipdds)

Zrdds <- DESeqDataSetFromMatrix(countData = ribozero.counts, colData = Ribozero, design = ~ Fetal + Zone)
Zrdds <- DESeq(Zrdds)
Zrres <- results(Zrdds)
Zpdds <- DESeqDataSetFromMatrix(countData = polya.counts, colData = PolyA, design = ~ Fetal + Zone)
Zpdds <- DESeq(Zpdds)
Zpres <- results(Zpdds)

Ap.dds <- DESeqDataSetFromMatrix(countData = Adult.polyA.counts, colData = Adult.polyA, design = ~ Zone)
Ap.dds <- DESeq(Ap.dds)
Apres <- results(Ap.dds)
Fp.dds <- DESeqDataSetFromMatrix(countData = Fetal.polyA.counts, colData = Fetal.polyA, design = ~ Zone)
Fp.dds <- DESeq(Fp.dds)
Fpres <- results(Fp.dds)
Ar.dds <- DESeqDataSetFromMatrix(countData = Adult.Ribo.counts, colData = Adult.Ribo, design = ~ Zone)
Ar.dds <- DESeq(Ar.dds)
Arres <- results(Ar.dds)
Fr.dds <- DESeqDataSetFromMatrix(countData = Fetal.Ribo.counts, colData = Fetal.Ribo, design = ~ Zone)
Fr.dds <- DESeq(Fr.dds)
Frres <- results(Fr.dds)

#How many genes are differentially expressed by Age?
Ager.dds <- DESeqDataSetFromMatrix(countData = ribozero.counts, colData = Ribozero, design = ~ Zone + Fetal)
Ager.dds <- DESeq(Ager.dds)
Agerres <- results(Ager.dds)
resultsNames(Ager.dds)
Agep.dds <- DESeqDataSetFromMatrix(countData = polya.counts, colData = PolyA, design = ~ Zone + Fetal)
Agep.dds <- DESeq(Agep.dds)
Agepres <- results(Agep.dds)

Cp.dds <- DESeqDataSetFromMatrix(countData = Cyt.polyA.counts, colData = Cyt.polyA, design = ~ Fetal)
Cp.dds <- DESeq(Cp.dds)
Cpres <- results(Cp.dds)
Np.dds <- DESeqDataSetFromMatrix(countData = Nuc.polyA.counts, colData = Nuc.polyA, design = ~ Fetal)
Np.dds <- DESeq(Np.dds)
Npres <- results(Np.dds)
Cr.dds <- DESeqDataSetFromMatrix(countData = Cyt.Ribo.counts, colData = Cyt.Ribo, design = ~ Fetal)
Cr.dds <- DESeq(Cr.dds)
Crres <- results(Cr.dds)
Nr.dds <- DESeqDataSetFromMatrix(countData = Nuc.Ribo.counts, colData = Nuc.Ribo, design = ~ Fetal)
Nr.dds <- DESeq(Nr.dds)
Nrres <- results(Nr.dds)

#Gene expression by library
Ln.dds <- DESeqDataSetFromMatrix(countData = Nucleus.counts, colData = Nucleus, design = ~ Fetal + Library)
Ln.dds <- DESeq(Ln.dds)
Lnres <- results(Ln.dds)

La.dds <- DESeqDataSetFromMatrix(countData = Adult.counts, colData = Adult, design = ~ Zone + Library)
La.dds <- DESeq(La.dds)
Larres <- results(La.dds)
Lf.dds <- DESeqDataSetFromMatrix(countData = Fetal.counts, colData = Fetal, design = ~ Zone + Library)
Lfp.dds <- DESeq(Lf.dds)
Lfpres <- results(Lf.dds)
Lc.dds <- DESeqDataSetFromMatrix(countData = Cytosol.counts, colData = Cytosol, design = ~ Fetal + Library)
Lc.dds <- DESeq(Lc.dds)
Lcrres <- results(Lc.dds)
Ln.dds <- DESeqDataSetFromMatrix(countData = Nucleus.counts, colData = Nucleus, design = ~ Fetal + Library)
Lnp.dds <- DESeq(Ln.dds)
Lnpres <- results(Ln.dds)

Ac.dds <- DESeqDataSetFromMatrix(countData = Adult.cyt.counts, colData = Adult.cyt, design = ~ Library)
Ac.dds <- DESeq(Ac.dds)
Acres <- results(Ac.dds)
Fc.dds <- DESeqDataSetFromMatrix(countData = Fetal.cyt.counts, colData = Fetal.cyt, design = ~ Library)
Fc.dds <- DESeq(Fc.dds)
Fcres <- results(Fc.dds)
An.dds <- DESeqDataSetFromMatrix(countData = Adult.nuc.counts, colData = Adult.nuc, design = ~ Library)
An.dds <- DESeq(An.dds)
Anres <- results(An.dds)
Fn.dds <- DESeqDataSetFromMatrix(countData = Fetal.nuc.counts, colData = Fetal.nuc, design = ~ Library)
Fn.dds <- DESeq(Fn.dds)
Fnres <- results(Fn.dds)